pkg/runtime/runtime.go

```diff
@@ -31,7 +31,9 @@ func startSyncer(ctx context.Context, notifier chan sync.INotify, syncr sync.ISy
 		log.Error(err)
 	}
 
-	go syncr.Notify(ctx, notifier)
+	ready := make(chan struct{})
```

Comment:
Can you outline the purpose of this? It's blocking the notifiers

Reply:
Added comments clarifying what this does: 
- https://github.com/open-feature/flagd/pull/85/commits/7975e4a3d371a96c56107094043cb6cf3acf94d7#diff-e56f9bb947923c6f8bb4c995bf4b9469a0ed117276223d16b177dcb04399a4bcR36
- https://github.com/open-feature/flagd/pull/85/commits/7975e4a3d371a96c56107094043cb6cf3acf94d7#diff-3dd5e32262f4cd7ce7746cc313911ec3bd30069d8a44ddcf4c2b1a7ae4ebc009R11

The runtime synchronise mechanism doesn't care about whether `Notify` is ready to emit events or not but without this the testing is forced to `time.Sleep` to wait for `Notify` to be ready. More info here: https://github.com/open-feature/flagd/pull/85#discussion_r930155185

Reply:
I would be open to having a return channel on the `Notify` method rather than you having to add this into the runtime. WDYT?

Reply:
I'm not following, are you suggesting changing `Notify` to return a channel?

Reply:
Yes that's what I'm thinking, or as a parameter.

Reply:
`ready chan<- struct{}` has been included as a parameter of `Notify` as a part of this PR.
Returning the channel instead wouldn't work as we need the channel prior to `Notify` returning.

Reply:
Why create this new channel rather than using the existing channel?

E.g. modifying the struct
```
w <- &Notifier{
					Event: Event[DefaultEventType]{
						Ready: true,
					},
				}
```

What do you think, wouldn't that keep it a bit cleaner?

Reply:
I've reworked the implementation to include a Ready type event on the notifier channel. What do you think?